python/lsst/pipe/supertask/cmdLineActivator.py

```diff
@@ -300,23 +358,20 @@ def makeTargetList(self, task_args):
 
         Returns
         -------
-        `list` of (args, kwargs) tuples which are arguments to be passed to
-        task execute() method.
+        `list` of (dataRefs, kwargs) tuples which are arguments to be passed
+        to `self._executeSuperTask()` method.
         """
-        refList = None
         kwargs = {}
         for optname, optval in vars(task_args).items():
             if isinstance(optval, pipeBaseArgParser.DataIdContainer):
-                if optname == "id":
-                    refList = optval.refList
-                else:
-                    kwargs[optname] = optval.refList
-        if refList is not None:
-            return [([ref], kwargs) for ref in refList]
-        else:
-            return [([], kwargs)]
+                kwargs[optname] = optval.refList
+        idRefList = kwargs.pop("id", None)
+
+        # forward collected dataRefs to task class
+        return self.task_class.makeTargetList(idRefList, kwargs)
```

Comment:
are you sure you mean `kwargs` and not `**kwargs`? If the former, I think the name is misleading, usually kwargs are expanded into the function you're calling?